Geometry/VeryForwardGeometry/python/geometryRPFromDB_cfi.py

```diff
@@ -1,12 +1,19 @@
 import FWCore.ParameterSet.Config as cms
 
 XMLIdealGeometryESSource_CTPPS = cms.ESProducer("XMLIdealGeometryESProducer",
-                                                rootDDName = cms.string('cms:CMSE'),
-                                                label = cms.string('CTPPS'),
-                                                appendToDataLabel = cms.string('XMLIdealGeometryESSource_CTPPS')
-                                                )
+    rootDDName = cms.string('cms:CMSE'),
+    label = cms.string('CTPPS'),
+    appendToDataLabel = cms.string('XMLIdealGeometryESSource_CTPPS')
+)
 
 ctppsGeometryESModule = cms.ESProducer("CTPPSGeometryESModule",
     verbosity = cms.untracked.uint32(1),
+    isRun2 = False,
```

Comment:
The definition of the ESProducer configuration needs the explicit type
```suggestion
    isRun2 = cms.bool(False),
```

Reply:
Many many thanks, @makortel! I was really starting to lose my sisu on this. So it seems the explicit type is indeed required for some parts. Should it also be replaced in the eras modifiers for safety? Or does https://github.com/cms-sw/cmssw/pull/31484#discussion_r533423845 still stand?

Reply:
> So it seems the explicit type is indeed required for some parts.

The explicit types are needed in the definition of a module or `PSet`.

> Should it also be replaced in the eras modifiers for safety?

No, in all customizations of an existing module where an existing parameter is modified  (see below) it is better to leave out the explicit type (framework ensures that the parameter really exists and thus protects for e.g. typos, also less typing andrea ding)
```py
# direct override
ctppsGeometryESModule.isRun2 = False

# clone
anotherName = ctppsGeometryESModule.clone(isRun2 = False)

# modifier
someModifier.toModify(ctppsGeometryESModule, isRun2 = False)
```

In other words,

> Or does [#31484 (comment)](https://github.com/cms-sw/cmssw/pull/31484#discussion_r533423845) still stand?

yes, my earlier comment still stands.

> I was really starting to lose my sisu on this

Keep it up!

Reply:
Thanks a lot for the very detailed explanation! Much appreciated!